### PR TITLE
[Feature] add more log info

### PR DESCRIPTION
**Pull request by YanhuiDua**
### Key Features & Changes

1. Configurable Initial Evaluation:
- “enable_initial_evaluate” has been added to the configuration. Users can now set this to False to skip first evaluation step before training begins.

2. Add timer for RLTrainer and Dataflow:
- **Add a new dependency, codetiming**, to precisely measure the execution time of different components within the RL pipeline.
- Logs now include detailed timing information for: Average processing time per sample in the DataFlow. Execution time for major functions within the RLTrainer.

3. Extensible Model Forward Pass:
- The **model.forward method now returns an extra_info dictionary**.
- This dictionary is designed to be extensible, allowing users to easily add custom info. Initially, it includes max_ratio and log_rank_loss for advanced reward modeling and analysis.

4. Unified and Enriched Logging:

- Consolidated Log Files: The logging format has been simplified. All logs for a specific rank (both training and rollout) are now consolidated into a single file, rank_{get_rank()}.log, eliminating the previous separation of train_0.log and rank_0.log.
- Trajectory File Summaries: Generated trajectory files now include a summary header containing key statistics about the rollouts, such as average, maximum, and minimum output length and reward scores.

### CODE REVIEW

File: xtuner/v1/datasets/config.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def build_datasets(
             _dataset = _dataset_config.build(_tokenize_fn)
             if get_rank() == 0:
                 logger.info(
-                    f"[Dataset] (Original) {_dataset_config.name}/{os.path.basename(anno_path)}: {len(_dataset)} samples."
+                    f"[Datasets] (Original) {_dataset_config.name}/{os.path.basename(anno_path)}: {len(_dataset)} samples."
```
Comment (hhaAndroid) [inline xtuner/v1/datasets/config.py]: 不用改

File: xtuner/v1/model/dense/dense.py [hunk 2]
```diff
@@ -89,9 +89,11 @@ def forward(
 
         hidden_states = self.norm(hidden_states)
 
-        loss, logits = self.lm_head(hidden_states, loss_ctx)  # type: ignore
+        loss, (logits, max_ratio) = self.lm_head(hidden_states, loss_ctx)
```
Comment (RangiLyu) [inline xtuner/v1/model/dense/dense.py]: head这里hard code有个max_ratio感觉不太合适啊，不是所有loss都求ratio，以及其他loss如果要统计别的信息，要怎么传出来？ @hhaAndroid

File: xtuner/v1/ray/rollout/worker.py [hunk 3]
```diff
@@ -283,19 +283,16 @@ async def rollout_task(
             # todo: test streaming infer with return_token_ids
             extra_params["stream"] = False
             if self.print_flag:
-                self.logger.warning("streaming infer is not supported when return_token_ids is True")
+                self.logger.warning("Streaming infer is not supported when return_token_ids is True")
+                if os.environ.get("XTUNER_USE_LMDEPLOY", "0") == "1":
+                    self.logger.warning("You should use lmdeploy main branch > v0.10.1 to support return_token_ids")
```
Comment (RangiLyu) [inline xtuner/v1/ray/rollout/worker.py]: 这里需要检查lmdeploy版本再报warning，装了正确版本就不报了

File: xtuner/v1/model/dense/dense.py (hunk 2 above)
Comment (YanhuiDua) [reply xtuner/v1/model/dense/dense.py]: 这个已经在改了，ModelOutput会新增一个字段为extra_info，用来包含其他例如“max_ratio“的字段

File: xtuner/v1/ray/rollout/worker.py (hunk 3 above)
Comment (YanhuiDua) [reply xtuner/v1/ray/rollout/worker.py]: ok, thanks
